### PR TITLE
feat(docs): add auto-docs creator

### PR DESCRIPTION
**Pull request by scottnath**
### Summary

Sets up the auto-docs package

@todo - documentation for other packages

### Added

- port of PAL JSDocs functionality, including typedef expansion plugin

### CODE REVIEW

File: packages/docs/cli.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+const jsdocConfig = require('./jsdoc-conf');
+
+/**
+ * Creates a markdown file containing documentation rendered from a set of files' JSDoc content.
+ *    Other than `data.dest` all other configurations are documented in the
+ *    [jsdoc2md API docs](https://github.com/jsdoc2md/jsdoc-to-markdown/blob/master/docs/API.md#jsdoc2mdrenderoptions--promise)
+ * @param {object} [data]
+ * @param {string} [data.dest] - file where docs should be rendered
+ *    for the repository we are cloning (see typedef)
```
Comment (elizabethsjudd) [inline packages/docs/cli.js]: Copy and paste error? `repository we are cloning`

File: packages/docs/cli.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * @fileoverview Command line interface (CLI) for this module
+ * @module docs/cli
+ */
+const fs = require('fs');
+const meow = require('meow');
+const jsdoc2md = require('jsdoc-to-markdown');
+
+const jsdocConfig = require('./jsdoc-conf');
```
Comment (elizabethsjudd) [inline packages/docs/cli.js]: Looks like you import this file but then don't use it. You only reference it's file name on line 36.

File: packages/docs/cli.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  }
+}
+
+/**
+ * Sets up the command line interface (CLI) for the module
+ * See [meow docs](https://github.com/sindresorhus/meow) for usage info
+ */
+const cli = meow(`
+Usage
+  $ Auto docs <action>
```
Comment (elizabethsjudd) [inline packages/docs/cli.js]:
```suggestion
  $ auto-docs <action>
```

File: packages/docs/lib/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @fileoverview Helpers for creating and processing JSDoc content
+ * @module docs
+ */
+
```
Comment (elizabethsjudd) [inline packages/docs/lib/index.js]: This file doesn't have anything in it. Do we need to keep it?

**Review by elizabethsjudd — COMMENTED** [review]
Kind of confusing having the package name be docs because typically we use that as the name for our directory that holds all documentation.  I would expect `api.md` to be in `packages/docs/docs/api.md`.  I wonder if the directory should be `auto-docs` to help with clarification.

The actual output of `api.md` is also hard to follow but I did see that the expander worked which seems to be the bulk of what this code is suppose to be doing.

File: packages/docs/cli.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+/**
+ * @fileoverview Command line interface (CLI) for this module
+ * @module docs/cli
+ */
+const fs = require('fs');
+const meow = require('meow');
+const jsdoc2md = require('jsdoc-to-markdown');
+
+const jsdocConfig = require('./jsdoc-conf');
+
```
Comment (scottnath) [inline packages/docs/cli.js]:
```suggestion
```

File: packages/docs/cli.js (hunk 1 above)
Comment (scottnath) [inline packages/docs/cli.js]:
```suggestion
 * @param {string} [data.dest] - full path to destination file for rendered doc
```

File: packages/docs/lib/index.js (hunk 4 above)
Comment (scottnath) [reply packages/docs/lib/index.js]: nah, will delete

**Comment by scottnath** [comment]
Thanks @elizabethsjudd ready for re-review please

**Comment by scottnath** [comment]
@elizabethsjudd changed to `auto-jsdocs`

ready for re-review

File: packages/auto-jsdocs/cli.js [hunk 6]
```diff
@@ -79,7 +79,7 @@ Usage
   $ auto-docs <action>
```
Comment (elizabethsjudd) [inline packages/auto-jsdocs/cli.js:79]:
```suggestion
  $ auto-jsdocs <action>
```

**Review by elizabethsjudd — APPROVED** [review]
One small documentation instance was missed but otherwise looks good
